pkg/kubectl/describe.go

```diff
@@ -1779,10 +1779,53 @@ func (d *ServiceAccountDescriber) Describe(namespace, name string, describerSett
 		}
 	}
 
-	return describeServiceAccount(serviceAccount, tokens)
+	existingPullSecrets := map[string]api.Secret{}
+	pullSecretListErr := false
+
+	// append opaque secrets to pull secrets list
+	opaqueSecretSelector := fields.SelectorFromSet(map[string]string{api.SecretTypeField: string(api.SecretTypeOpaque)})
+	opaqueSecrets, err := d.Core().Secrets(namespace).List(api.ListOptions{
+		FieldSelector: opaqueSecretSelector,
+	})
+	if err == nil {
+		for _, s := range opaqueSecrets.Items {
+			existingPullSecrets[s.GetName()] = s
+		}
+	} else {
+		pullSecretListErr = true
+	}
+
+	//append dockercfg secrets to pull secrets list
+	dockercfgSecretSelector := fields.SelectorFromSet(map[string]string{api.SecretTypeField: string(api.SecretTypeDockercfg)})
+	dockercfgSecrets, err := d.Core().Secrets(namespace).List(api.ListOptions{
+		FieldSelector: dockercfgSecretSelector,
+	})
+	if err == nil {
+		for _, s := range dockercfgSecrets.Items {
+			existingPullSecrets[s.GetName()] = s
+		}
+	} else {
+		pullSecretListErr = true
+	}
+
+	// add all secret names in service account.
+	// Append extra information to deleted secrets
+	saPullSecrets := []string{}
+	for _, s := range serviceAccount.ImagePullSecrets {
+		// if an error occurred while listing any secrets or the
+		// secret exists, add secret without any additional information
+		if secret, exists := existingPullSecrets[s.Name]; pullSecretListErr || exists {
+			saPullSecrets = append(saPullSecrets, secret.Name)
+			continue
+		}
+		sName := fmt.Sprintf("%s (%s)", s.Name, "not found")
+		saPullSecrets = append(saPullSecrets, sName)
+	}
+
+	return describeServiceAccount(serviceAccount, tokens, saPullSecrets)
```

Comment:
Just pass in the list of secrets found in the namespace (and some indication if the list couldn't be loaded, a nil slice, or a bool flag or something).

Modify the printing of the secret name in describeServiceAccount to tack on "(not found)" or something if the list of secrets in the namespace was successfully loaded and does not contain the secret name